packages/plugins/typescript-react-apollo/src/index.ts

```diff
@@ -4,31 +4,31 @@ import { flattenTypes } from 'graphql-codegen-plugin-helpers';
 import { GraphQLSchema } from 'graphql';
 import * as Handlebars from 'handlebars';
 import * as rootTemplate from './root.handlebars';
-import { generateFragments, gql, propsType, shouldOutputHook, gqlImport } from './helpers';
+import { generateFragments, gql, propsType, gqlImport } from './helpers';
 import { extname } from 'path';
 
 export interface TypeScriptReactApolloConfig extends TypeScriptCommonConfig {
   noGraphqlTag?: boolean;
   noNamespaces?: boolean;
-  noHOC?: boolean;
-  noComponents?: boolean;
-  withHooks?: boolean;
-  withSubscriptionHooks?: boolean;
+  withHOC?: boolean;
+  withComponents?: boolean;
+  withHooks?: boolean | string;
   gqlImport?: string;
 }
 
 export const plugin: PluginFunction<TypeScriptReactApolloConfig> = async (
   schema: GraphQLSchema,
   documents: DocumentFile[],
-  config: TypeScriptReactApolloConfig
+  config: TypeScriptReactApolloConfig = {}
 ): Promise<string> => {
+  config.withHOC = 'withHOC' in config ? config.withHOC : true;
```

Comment:
Just a personal preference, probably not a big different
```suggestion
  config.withHOC = config.withHOC !== undefined ? config.withHOC : true;
```